darts/utils/timeseries_generation.py

```diff
@@ -675,11 +675,21 @@ def datetime_attribute_timeseries(
     if one_hot:
         values_df = pd.get_dummies(values)
         # fill missing columns (in case not all values appear in time_index)
-        attribute_range = range(num_values_dict[attribute])
-        for i in attribute_range:
-            if i not in values_df.columns:
-                values_df[i] = 0
-        values_df = values_df[attribute_range]
+        attribute_range = np.arange(num_values_dict[attribute])
```

Comment:
In my opinion we could simply replace the original `values_df[i] = 0` with `values_df.loc[:, i] = 0` to get rid of the warning.
This test is not really computationally expensive, so we don't have to worry about performance here.